src/preact/base-element.js

```diff
@@ -513,10 +522,15 @@ function collectProps(Ctor, element, ref, defaultProps) {
   // Props.
   for (const name in propDefs) {
     const def = propDefs[name];
-    const value =
-      def.type == 'boolean'
-        ? element.hasAttribute(def.attr)
-        : element.getAttribute(def.attr);
+    let value;
+    if (def.attr) {
+      value =
+        def.type == 'boolean'
+          ? element.hasAttribute(def.attr)
+          : element.getAttribute(def.attr);
+    } else if (def.attrs) {
+      value = def.parseAttrs(element);
```

Comment:
We should check for `parseAttrs`, since we're calling it?
```suggestion
    } else if (def.parseAttrs) {
      value = def.parseAttrs(element);
```

Reply:
Agree. That's better. I also added `devAssert(def.attrs)` in this case to ensure we don't forget to add them. Otherwise mutation observer won't be able to catch all cases.